### PR TITLE
bdd: Create Invitation with Public DID

### LINKED ISSUE #519: BDD test for Create Invitation with Public DID
Add BDD test for creating invitation with Public DID

### PR DESCRIPTION
**Pull request by sandrask**
BDD test for Create Invitation with public DID

Closes #519

Signed-off-by: Sandra Vrtikapa <sandra.vrtikapa@securekey.com>

### CODE REVIEW

File: test/bdd/context.go [hunk 1]
```diff
@@ -23,15 +25,19 @@ type Context struct {
 
 // NewContext create new Context
 func NewContext() (*Context, error) {
-	instance := Context{DIDExchangeClients: make(map[string]*didexchange.Client),
-		Invitations: make(map[string]*didexchange2.Invitation), PostStatesFlag: make(map[string]map[string]chan bool),
-		ConnectionID: make(map[string]string), AgentCtx: make(map[string]*context.Provider)}
+	instance := Context{
+		DIDExchangeClients: make(map[string]*didexchange.Client),
+		Invitations:        make(map[string]*didexchange2.Invitation),
```
Comment (troyronda) [inline test/bdd/context.go]: Suggest renaming the import aliases - it's hard to tell from the code what is coming from where.

File: test/bdd/features/didexchange_public_did.feature [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# SPDX-License-Identifier: Apache-2.0
+#
+# Reference : https://github.com/hyperledger/aries-rfcs/tree/master/features/0023-did-exchange
+
+@all
+Feature: Decentralized Identifier(DID) exchange between the agents using public did in invitation
+
+  @didexchange_public_did
+  Scenario: did exchange e2e flow using public DID in invitation
+    Given "Maria" agent is running on "localhost" port "random" with http-binding did resolver url "http://localhost:48326/.sidetree/document" which accepts did method "sidetree"
```
Comment (troyronda) [inline test/bdd/features/didexchange_public_did.feature]: How do we know the port is `48326`?

File: test/bdd/features/didexchange_public_did.feature [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    And   "Maria" registers to receive notification for post state event "completed"
+    Given "Lisa" agent is running on "localhost" port "random" with http-binding did resolver url "http://localhost:48326/.sidetree/document" which accepts did method "sidetree"
+    And   "Lisa" registers to receive notification for post state event "completed"
+    And   "Maria" creates public DID using sidetree "http://localhost:48326/.sidetree/document"
+    # we wait until observer polls sidetree txn
+    Then  we wait 1 seconds
```
Comment (troyronda) [inline test/bdd/features/didexchange_public_did.feature]: This seem like we could flake the tests.

File: test/bdd/features/didexchange_public_did.feature (hunk 2 above)
Comment (troyronda) [reply test/bdd/features/didexchange_public_did.feature]: e.g., can this be an environment variable so it matches whatever the docker config is?

File: test/bdd/context.go (hunk 1 above)
Comment (troyronda) [reply test/bdd/context.go]: Also I don't understand why we are even importing from the protocol in a BDD test. It is supposed to be completely wrapped by DID exchange client.

@fqutishat @rolsonquadras @sandrask

Comment (rolsonquadras) [reply test/bdd/context.go]: Issue https://github.com/hyperledger/aries-framework-go/issues/530 to fix this.

File: test/bdd/did_exchange_steps.go [hunk 4]
```diff
@@ -39,6 +39,20 @@ func (d *DIDExchangeSteps) createInvitation(inviterAgentID string) error {
 	return nil
 }
 
+func (d *DIDExchangeSteps) createInvitationWithDID(inviterAgentID string) error {
+	invitation, err := d.bddContext.DIDExchangeClients[inviterAgentID].CreateInvitationWithDID(inviterAgentID, d.bddContext.PublicDIDs[inviterAgentID].ID)
+	if err != nil {
+		return fmt.Errorf("failed to create invitation: %w", err)
```
Comment (troyronda) [inline test/bdd/did_exchange_steps.go]: Just a thought that I am posting everywhere I see wrapping error strings that have "failed to" in them :). https://github.com/uber-go/guide/blob/master/style.md#error-wrapping

File: test/bdd/features/didexchange_public_did.feature (hunk 2 above)
Comment (sandrask) [reply test/bdd/features/didexchange_public_did.feature]: Created new issue #532

File: test/bdd/features/didexchange_public_did.feature (hunk 3 above)
Comment (sandrask) [reply test/bdd/features/didexchange_public_did.feature]: I will add polling

File: test/bdd/did_exchange_steps.go (hunk 4 above)
Comment (sandrask) [reply test/bdd/did_exchange_steps.go]: Will fix

**Pull request merged**